lib/cloud/azure/interfaces.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/mysql/armmysql"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/postgresql/armpostgresql"
+)
+
+// DBServersClient provides an interface for fetching Azure DB Servers.
+type DBServersClient interface {
+	// ListAll returns all Azure DB servers within an Azure subscription.
+	ListAll(ctx context.Context, maxPages int) ([]*DBServer, error)
+	// ListWithinGroup returns all Azure DB servers within an Azure resource group.
+	ListWithinGroup(ctx context.Context, group string, maxPages int) ([]*DBServer, error)
```

Comment:
nit: Is `maxPages` necessary?

Reply:
Are you asking if it's necessary to pass in as an argument instead of making it a constant within `lib/cloud/azure`?

Reply:
It seems the max number of database servers per subscription is pretty small. I don't think we will ever hit 10 pages. 

Also, it's unclear from this interface (or Azure SDK) how many items per page. When using an interface like this, if I pass in 2 or 20, I am not sure what I should expect.

I don't mind keeping it in the interface too.

Reply:
I'm actually not sure what the limit is per subscription. Reading [subscription limits](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/azure-subscription-service-limits#subscription-limits) I see you can have up to 980 resource groups per subscription, and 800 of a resource type within each group, and looking at [resources-without-resource-group-limit](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resources-without-resource-group-limit) MySQL and Postgres single-server are exempt from the 800 limit.

I also see no mention of how many items there can be per page. However, looking at the `NewListPager` and `pager.NextPage` definitions in armmysql and armpostgresql, it looks like the SDK calling this a "pager" with for-loop examples that iterate over "pages" was very deceptive 😐 . No paging is really happening with this thing. It will always return one page. I guess I can remove the maxPages then.

Reply:
I think you are right. I was wrong about the limit on the max number of database servers. I was looking at the limit for [Azure SQL](https://docs.microsoft.com/en-us/azure/azure-sql/database/resource-limits-logical-server?view=azuresql). 

Thanks for updating.